In [1]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import seaborn as sns

In [2]:
df_sachin = pd.read_csv('Sachin_ODI.csv') 

In [4]:
df_sachin.head(5)

runs  NotOut mins  bf  fours  sixes      sr  Inns           Opp    Ground  \
0    13       0   30  15      3      0   86.66     1   New Zealand    Napier   
1    37       0   75  51      3      1   72.54     2  South Africa  Hamilton   
2    47       0   65  40      7      0  117.50     2     Australia   Dunedin   
3    48       0   37  30      9      1  160.00     2    Bangladesh   Sharjah   
4     4       0   13   9      1      0   44.44     2      Pakistan   Sharjah   

         Date        Winner    Won  century  
0  1995-02-16   New Zealand  False    False  
1  1995-02-18  South Africa  False    False  
2  1995-02-22         India   True    False  
3  1995-04-05         India   True    False  
4  1995-04-07      Pakistan  False    False

## What is the Probability of Wining?

In [5]:
df_sachin["Won"].value_counts()

Won
True     184
False    176
Name: count, dtype: int64

In [11]:
prob_wining = 184/(184+176)
prob_wining

0.5111111111111111

In [14]:
df_won = df_sachin[df_sachin["Won"]==1]
len(df_won)

184

In [13]:
len(df_sachin)

360

In [16]:
prob_wining = len(df_won)/len(df_sachin)
round(prob_wining,2)

0.51

## What is the prorbability of Centuries? 

In [20]:
no_of_sachin_cen = len(df_sachin[df_sachin['century'] == True]) 
no_of_sachin_cen

46

In [22]:
prob_century = no_of_sachin_cen/len(df_sachin)
round(prob_century,2)

0.13

## Wining and Centuries

In [23]:
df_sachin[["century","Won"]].value_counts()

century  Won  
False    False    160
         True     154
True     True      30
         False     16
Name: count, dtype: int64

In [48]:
df_sachin[["century","Won"]].value_counts().unstack()

Won      False  True 
century              
False      160    154
True        16     30

In [24]:
# Contingency Table
pd.crosstab(index=df_sachin["century"],columns=df_sachin["Won"],margins=True)

Won      False  True  All
century                  
False      160   154  314
True        16    30   46
All        176   184  360

### What is the probability of winning and scoring a century in a match?
1) In how many matches has winning and century happened? 30
2) Totally how many matches are there? 360
3) What is the probability of winning and century? 30/360

In [25]:
df_win_cen = df_sachin[(df_sachin["century"]==True) & (df_sachin["Won"]==True)]

In [26]:
len(df_win_cen)

30

In [27]:
pro_win_and_cen = len(df_win_cen)/len(df_sachin)
pro_win_and_cen

0.08333333333333333

## Wining OR Century
1) In how many matches will winning or century happen?

a) 184 + 46 

b) 184 + 16

c) 154 + 16

d) 184 + 46 - 30

In [28]:
184 + 46 - 30

200

In [29]:
df_win_or_century = df_sachin[(df_sachin["century"] == True) | (df_sachin["Won"] == True)]

In [30]:
len(df_win_or_century)

200

In [31]:
prob_win_or_century = len(df_win_or_century) / len(df_sachin)
prob_win_or_century

0.5555555555555556

In [32]:
200/360

0.5555555555555556

In [37]:
prob_in_or_loss = prob_wining + prob_century - pro_win_and_cen
prob_in_or_loss

0.5555555555555555

## Complement

In [34]:
# Win or Loss
df_won = df_sachin[df_sachin["Won"]==1]
prob_won = len(df_won)/len(df_sachin)
prob_won

0.5111111111111111

In [35]:
df_loss = df_sachin[df_sachin["Won"]==0]
prob_loss = len(df_loss)/len(df_sachin)
prob_loss

0.4888888888888889

In [36]:
total_prob = prob_won + prob_loss
total_prob

1.0

In [38]:
# At least 8 fours are scored
#Complement: Atmost 7 fours are scored

In [43]:
df_atleast_8_fours = df_sachin[df_sachin["fours"] >= 8]
prob_atleast_8_fours = len(df_atleast_8_fours)/len(df_sachin)
prob_atleast_8_fours

0.26666666666666666

In [40]:
len(df_atleast_8_fours)

96

In [44]:
df_atmost_7_fours = df_sachin[df_sachin["fours"] < 8]
prob_atmost_7_fours = len(df_atmost_7_fours)/len(df_sachin)
prob_atmost_7_fours

0.7333333333333333

In [42]:
len(df_atmost_7_fours)

264

In [47]:
total_prob2 = prob_atleast_8_fours + prob_atmost_7_fours
total_prob2


1.0